discovery/store.go

```diff
@@ -291,45 +245,13 @@ func (s *sqlStore) get(serviceID string, tag *Tag) ([]vc.VerifiablePresentation,
 // Wildcard matching is supported by prefixing or suffixing the value with an asterisk (*).
 // It returns the presentations which contain credentials that match the given query.
 func (s *sqlStore) search(serviceID string, query map[string]string) ([]vc.VerifiablePresentation, error) {
-	propertyColumns := map[string]string{
-		"id":                   "cred.credential_id",
-		"issuer":               "cred.credential_issuer",
-		"type":                 "cred.credential_type",
-		"credentialSubject.id": "cred.credential_subject_id",
-	}
-
 	stmt := s.db.Model(&presentationRecord{}).
 		Where("service_id = ?", serviceID).
-		Joins("inner join discovery_credential cred ON cred.presentation_id = discovery_presentation.id")
-	numProps := 0
-	for jsonPath, value := range query {
-		if value == "*" {
-			continue
-		}
-		// sort out wildcard mode: prefix and postfix asterisks (*) are replaced with %, which then is used in a LIKE query.
-		// Otherwise, exact match (=) is used.
-		var eq = "="
-		if strings.HasPrefix(value, "*") {
-			value = "%" + value[1:]
-			eq = "LIKE"
-		}
-		if strings.HasSuffix(value, "*") {
-			value = value[:len(value)-1] + "%"
-			eq = "LIKE"
-		}
-		if column := propertyColumns[jsonPath]; column != "" {
-			stmt = stmt.Where(column+" "+eq+" ?", value)
-		} else {
-			// This property is not present as column, but indexed as key-value property.
-			// Multiple (inner) joins to filter on a dynamic number of properties to filter on is not pretty, but it works
-			alias := "p" + strconv.Itoa(numProps)
-			numProps++
-			stmt = stmt.Joins("inner join discovery_credential_prop "+alias+" ON "+alias+".credential_id = cred.id AND "+alias+".path = ? AND "+alias+".value "+eq+" ?", jsonPath, value)
-		}
-	}
+		Joins("inner join discovery_credential ON discovery_credential.presentation_id = discovery_presentation.id")
+	stmt = store.CredentialStore{}.BuildSearchStatement(stmt, "discovery_credential.credential_id", query)
 
 	var matches []presentationRecord
-	if err := stmt.Find(&matches).Error; err != nil {
+	if err := stmt.Preload("Credentials").Preload("Credentials.Credential").Find(&matches).Error; err != nil {
```

Comment:
THe properties aren't returned, so need to preload it. I don't think we can add the preload to `BuildSearchStatement`, since it appears to refer to the tables to preload from the root table.